node/bft/src/worker.rs

```diff
@@ -508,12 +509,25 @@ impl<N: Network> Worker<N> {
             );
         }
         // Wait for the transmission to be fetched.
-        match timeout(Duration::from_millis(MAX_FETCH_TIMEOUT_IN_MS), callback_receiver).await {
-            // If the transmission was fetched, return it.
-            Ok(result) => Ok((transmission_id, result?)),
-            // If the transmission was not fetched, return an error.
-            Err(e) => bail!("Unable to fetch transmission - (timeout) {e}"),
-        }
+
```

Comment:
nit: Is this really cleaner? This seems harder to read.

You could also pre-declare `fmt_id(transmission_id.checksum().unwrap_or_default()` and `fmt_id(transmission_id)` for cleanliness